src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -0,0 +1,156 @@
+// Libraries
+import React, {FC, useCallback, useState} from 'react'
+import {fromFlux} from '@influxdata/giraffe'
+import {useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
+
+// Utils
+import {postWrite} from 'src/client'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+import {getByID} from 'src/resources/selectors'
+
+// Types
+import {AppState, Bucket, ResourceType, WritePrecision} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface CsvUploaderContextType {
+  hasFile: boolean
+  progress: number
+  uploadCsv: (csv: string) => void
+  uploadFinished: boolean
+}
+
+export const DEFAULT_CONTEXT: CsvUploaderContextType = {
+  hasFile: false,
+  progress: 0,
+  uploadCsv: (_: string) => {},
+  uploadFinished: false,
+}
+
+export const CsvUploaderContext = React.createContext<CsvUploaderContextType>(
+  DEFAULT_CONTEXT
+)
+
+const MAX_CHUNK_SIZE = 1750
+
+export const CsvUploaderProvider: FC<Props> = React.memo(({children}) => {
+  const [progress, setProgress] = useState(0)
+  const {bucketID} = useParams()
+  const [hasFile, setHasFile] = useState(false)
+  const [uploadFinished, setUploadFinished] = useState(false)
```

Comment:
not a big deal, but isn't hasFile and uploadFinished the same as RemoteDataState (loading and done respectively). could join those two if it makes sense

Reply:
It'd be more apt to say `NotStarted` and `Done` but considering it's a binary outcome, might be quicker to use a bool rather than an enum?

Reply:
hasFile = false => notstarted
hasFile = true => loading
uploadFinished = true => done

Reply:
Aw, I see what you're saying. Totally can consolidate those